front/app/components/ConsentManager/index.tsx

```diff
@@ -33,10 +39,34 @@ const ConsentManager = () => {
   const [cookieConsent, setCookieConsent] = useState<IConsentCookie | null>(
     null
   );
-  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [screen, setScreen] = useState<'initial' | 'preferences' | null>(null);
+  const [searchParams] = useSearchParams();
+  const from = searchParams.get('from');
 
   const { data: appConfiguration } = useAppConfiguration();
   const { data: authUser } = useAuthUser();
+  const activeDestinations = getActiveDestinations(
+    appConfiguration?.data,
+    authUser?.data
+  );
+  const activeCategorizedDestinations = categorizeDestinations(
+    appConfiguration?.data,
+    activeDestinations
+  );
+  const isConsentRequired = getConsentRequired(
+    cookieConsent,
+    activeDestinations
+  );
+
+  useEffect(() => {
+    if (from === 'cookie-modal') {
```

Comment:
When we click the link to the cookie policy in the modal, we wouldn't be able to read the cookie policy without this, because the modal is on top of the page.